https://github.com/zephyrproject-rtos/zephyr/pull/70000/checks?check_run_id=22476480950 show the expected `printf` failures and shift width failures that motivated this new test in the first place. Examples below.

```
__w/zephyr/zephyr/subsys/llext/llext.c:160:25: error: format '%x' expects argument of type 'unsigned int', but argument 3 has type 'elf64_off' {aka 'long long unsigned int'} [-Werror=format=]
  160 |                 LOG_DBG("section %d at %x: name %d, type %d, flags %x, addr %x, size %d",
      |                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  161 |                         i,
  162 |                         ldr->hdr.e_shoff + i * ldr->hdr.e_shentsize,
      |                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                                          |
      |                                          elf64_off {aka long long unsigned int}
```

```
/__w/zephyr/zephyr/include/zephyr/llext/elf.h:349:29: error: right shift count >= width of type [-Werror=shift-count-overflow]
  349 | #define ELF64_R_SYM(i) ((i) >> 32)
      |                             ^~

```

--- a/tests/subsys/llext/simple/CMakeLists.txt
+++ b/tests/subsys/llext/simple/CMakeLists.txt
@@ -15,6 +15,7 @@ target_include_directories(app PRIVATE
   ${ZEPHYR_BASE}/arch/${ARCH}/include
 )
 
+if(NOT LOADER_BUILD_ONLY)
 # generate extension targets foreach extension given by name
 foreach(ext_name hello_world logging relative_jump)
   set(ext_src ${PROJECT_SOURCE_DIR}/src/${ext_name}_ext.c)
@@ -26,3 +27,4 @@ foreach(ext_name hello_world logging relative_jump)
   )
   generate_inc_file_for_target(app ${ext_bin} ${ext_inc})
 endforeach()
+endif()

--- a/tests/subsys/llext/simple/src/test_llext_simple.c
+++ b/tests/subsys/llext/simple/src/test_llext_simple.c
@@ -120,6 +120,8 @@ void load_call_unload(struct llext_test *test_case)
 
 	llext_unload(&ext);
 }
+
+#ifndef LOADER_BUILD_ONLY
 /*
  * Attempt to load, list, list symbols, call a fn, and unload each
  * extension in the test table. This excercises loading, calling into, and
@@ -151,6 +153,7 @@ static LLEXT_CONST uint8_t relative_jump_ext[] __aligned(4) = {
 	#include "relative_jump.inc"
 };
 LLEXT_LOAD_UNLOAD(relative_jump, true)
+#endif /* ! LOADER_BUILD_ONLY */
 
 /*
  * Ensure that EXPORT_SYMBOL does indeed provide a symbol and a valid address

--- a/tests/subsys/llext/simple/testcase.yaml
+++ b/tests/subsys/llext/simple/testcase.yaml
@@ -6,7 +6,25 @@ common:
   platform_exclude:
     - numaker_pfm_m487 # See #63167
     - qemu_cortex_r5 # unsupported relocations
+
 tests:
+  # add_llext_target() supports a fairly limited number of
+  # CONFIG_urations.  For instance, invoking add_llext_target()
+  # currently blocks us from compiling subsys/llext/*.c in 64bits mode;
+  # CMake aborts before even invoking the compiler.
+  #
+  # While there is in practice no value in compiling subsys/llext/*.c
+  # without actually running it to load some extension, let's keep it in
+  # good shape and ready to be used when add_llext_target()
+  # limitations get lifted in the future.
+  llext.simple.loader_build:
+    build_only: true
+    # How to override the above and allow ANY arch?
+    arch_allow: arm arm64 x86 x86_64 xtensa posix
+    extra_args:
+      - LOADER_BUILD_ONLY=1
+      - EXTRA_CFLAGS=-DLOADER_BUILD_ONLY=1
+
   llext.simple.readonly:
     arch_exclude: xtensa # for now
     filter: not CONFIG_MPU and not CONFIG_MMU and not CONFIG_SOC_SERIES_S32ZE

